compliance_checker/cf/cf.py

```diff
@@ -2513,7 +2516,7 @@ def check_cell_boundaries(self, ds):
             if boundary_variable_name not in ds.variables:
                 valid = False
                 reasoning.append("Boundary variable {} referenced by {} not "
-                                 "found in dataset variables".format(boundary_variable.name,
+                                 "found in dataset variables".format(boundary_variable_name,
```

Comment:
It's likely that the tests don't capture the bad case which would cause this block to be executed.